core/stream/convert.go

```diff
@@ -290,29 +299,59 @@ func intCollapse(count int, dst, src buf) error {
 	sourceStream := binary.BitStream{Data: src.bytes, ReadPos: src.offset}
 	destStream := binary.BitStream{Data: dst.bytes, WritePos: dst.offset}
 	maxPossibleValue := math.Pow(2, float64(srcBitsExcSign)) - 1
+	maxWithSign := math.Pow(2, float64(srcBitsIncSign)) - 1
 	min, max := maxPossibleValue, float64(0)
 	if src.component.Channel == Channel_Depth {
 		for i := 0; i < count; i++ {
-			f := float64(sourceStream.Read(srcBitsExcSign)) / maxPossibleValue
+			f := float64(0)
+
+			if srcTy.Signed && !signed {
```

Comment:
This is a bit at the borderline of being nit but as we are not assigning any vaule to `signed` maybe it might be a good idea to change it `dstTy.Signed` for readablity. Then it's more clear that it's actually from signed to unsigned in one pass of reading.

Reply:
I agree. I only left it in there because I try to leave pre-existing stuff like that in there just in case.